modules/soc_smarteq/soc_smarteq.py

```diff
@@ -21,10 +23,29 @@
 GUID = "280C6B55-F179-4428-88B6-E0CCF5C22A7C"
 ACCEPT_LANGUAGE = "de-de"
 
+TOKENS_REFRESH_THRESHOLD = 3600
+
+
+# helper functions
+def keys_exist(element, *keys):
+    # Check if *keys (nested) exists in `element` (dict).
+    if not isinstance(element, dict):
+        raise AttributeError('keys_exist() expects dict as first argument - got type ' + str(type(element)))
+    if len(keys) == 0:
+        raise AttributeError('keys_exist() expects at least two arguments, one given.')
+
+    _element = element
+    for key in keys:
+        try:
+            _element = _element[key]
```

Comment:
Nach dem ersten Durchlauf ist `_element = _element[key1]`.
`key2` muss demnach in `_element[key1]` enthalten sein. `_element[key1][key2]`
Ist das so beabsichtigt oder war keine verschachtelte Prüfung geplant?
Wenn das Verhalten so richtig ist, würde ich die Methode umbenennen: `nested_keys_exist()`